### PR TITLE
AP-4688: Flow refactor partner employment steps

### PR DESCRIPTION
**Pull request by naseberry**
## What

[Flow refactoring - provider_partner - employment details steps](https://dsdmoj.atlassian.net/browse/AP-4688)

Refactor the employment details steps flow/steps into individual step classes.

Steps:

1. partner_employed
2. partner_use_ccms_employment
3. partner_full_employment_details
4. partner_employment_incomes
5. partner_unexpected_employment_income

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- The standards in the [Git Workflow document on Confluence](https://dsdmoj.atlassian.net/wiki/spaces/ATPPB/pages/4602855954/Git+Workflow) should be followed
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: spec/services/flow/steps/partner/full_employment_details_step_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+RSpec.describe Flow::Steps::Partner::FullEmploymentDetailsStep, type: :request do
+  let(:legal_aid_application) { create(:legal_aid_application) }
+
+  describe "#path" do
+    subject { described_class.path.call(legal_aid_application) }
+
+    it { is_expected.to eql providers_legal_aid_application_partners_full_employment_details_path(legal_aid_application) }
+  end
+
+  describe "#forward" do
```
Comment (colinbruce) [inline spec/services/flow/steps/partner/full_employment_details_step_spec.rb:12]: Did this step not have any controller redirect tests?

File: spec/services/flow/steps/partner/employment_incomes_step_spec.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+RSpec.describe Flow::Steps::Partner::EmploymentIncomesStep, type: :request do
+  let(:legal_aid_application) { create(:legal_aid_application) }
+
+  describe "#path" do
+    subject { described_class.path.call(legal_aid_application) }
+
+    it { is_expected.to eql providers_legal_aid_application_partners_employment_income_path(legal_aid_application) }
+  end
+
+  describe "#forward" do
```
Comment (colinbruce) [inline spec/services/flow/steps/partner/employment_incomes_step_spec.rb:12]: Ditto on request specs

File: spec/services/flow/steps/partner/unexpected_employment_income_step_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+RSpec.describe Flow::Steps::Partner::UnexpectedEmploymentIncomeStep, type: :request do
+  let(:legal_aid_application) { create(:legal_aid_application) }
+
+  describe "#path" do
+    subject { described_class.path.call(legal_aid_application) }
+
+    it { is_expected.to eql providers_legal_aid_application_partners_unexpected_employment_income_path(legal_aid_application) }
+  end
+
+  describe "#forward" do
```
Comment (colinbruce) [inline spec/services/flow/steps/partner/unexpected_employment_income_step_spec.rb:12]: Ditto on request specs

**Review by colinbruce — COMMENTED** [review]
There don't seem to be any updates to request specs on the final three steps, was this a choice?

File: spec/services/flow/steps/partner/employment_incomes_step_spec.rb (hunk 2 above)
Comment (naseberry) [reply spec/services/flow/steps/partner/employment_incomes_step_spec.rb:12]: This should now be resolved.

File: spec/services/flow/steps/partner/full_employment_details_step_spec.rb (hunk 1 above)
Comment (naseberry) [reply spec/services/flow/steps/partner/full_employment_details_step_spec.rb:12]: This should now be resolved.

File: spec/services/flow/steps/partner/unexpected_employment_income_step_spec.rb (hunk 3 above)
Comment (naseberry) [reply spec/services/flow/steps/partner/unexpected_employment_income_step_spec.rb:12]: Missing, I've now added it.

**Review by colinbruce — APPROVED** [review]
LGTM :+1:

**Pull request merged**